src/basic/hostname-util.c

```diff
@@ -11,12 +11,29 @@
 #include "env-file.h"
 #include "hostname-util.h"
 #include "os-util.h"
+#include "proc-cmdline.h"
 #include "string-util.h"
 #include "strv.h"
 
 char* get_default_hostname(void) {
         int r;
 
+        /* The hostname= kernel commandline parameter is read by the kernel and set as the initial hostname.
+         * Use it as the highest priority default.
+         * (Not to be confused with systemd.hostname=, which is considered a static hostname).
+         */
+        _cleanup_free_ char *d = NULL;
+        r = proc_cmdline_get_key("hostname", 0, &d);
+        if (r < 0)
+                log_warning_errno(r, "Failed to retrieve default hostname from kernel command line, ignoring: %m");
+        else if (r > 0) {
+                if (hostname_is_valid(d, true)) {
```

Comment:
uh, the 2nd argument is actually a flags argument, not a boolean one (i guess you copied this from the location this fixes → #28259, right?)

Reply:
> (i guess you copied this from the location this fixes → #28259, right?)

Yes indeed. That change also resulted in a conflict when rebasing, as I've moved the code that was changed in #28259.

I've decided to use `hostname_is_valid(h, 0)` here instead of `hostname_is_valid(h, VALID_HOSTNAME_TRAILING_DOT)`, because that's what the other checks in the function use.
Does that sound correct to you?